roles/network_plugin/cilium/templates/cilium/ds.yml.j2

```
@@ -96,7 +96,7 @@ spec:
               fieldPath: metadata.namespace
         - name: CILIUM_CLUSTERMESH_CONFIG
           value: /var/lib/cilium/clustermesh/
-{% if cilium_kube_proxy_replacement == 'strict' %}
+{% if cilium_kube_proxy_replacement == 'strict' or  cilium_kube_proxy_replacement == 'true' %}
```

thanks @cccsss01, `strict` mode has been deprecated since cilium 1.14+, now we use true or false to configure cilium kubeProxyReplacement, see https://github.com/cilium/cilium/blob/bd537a359dddfe695660befb9e5023e6aefe6440/install/kubernetes/cilium/values.yaml#L1542

Could you also mind fixing the PR title? It looks like the length is over the limit.
```suggestion
{% if cilium_kube_proxy_replacement == 'strict' or cilium_kube_proxy_replacement == 'true' %}
```

---

and also, can you add a comment to there? 
https://github.com/kubernetes-sigs/kubespray/blob/7a9def547ec02b6821d1e9d9a5ce3ba41f4ff5e6/roles/network_plugin/cilium/defaults/main.yml#L48 and https://github.com/kubernetes-sigs/kubespray/blob/7a9def547ec02b6821d1e9d9a5ce3ba41f4ff5e6/inventory/sample/group_vars/k8s_cluster/k8s-net-cilium.yml#L54